src/ui/public/vis/vis.js

```diff
@@ -94,10 +98,13 @@ export function VisProvider(Private, indexPatterns, getAppState) {
             const appState = getAppState();
             filterBarClickHandler(appState)(event);
           },
-          addFilter: (data, columnIndex, rowIndex) => {
+          addFilter: (data, columnIndex, rowIndex, cellValue) => {
```

Comment:
I am not a huge fan of that syntax, that we need to pass in a `rowIndex` even in cases we might not have it, but I see, that ` addFilter(data, { columnIndex, rowIndex, cellValue })` might also not be the best solution (especially since we know we need the `columnIndex` and `addFilter(data, columnIndex, { rowIndex, cellValue })` for sure looks even more weird...

So just wanted to point that out, even without having a proper good solution (but maybe @markov00 or so has an idea).

Could we at least give `cellValue = null` a default value?

Reply:
As written few seconds ago on my review, are we sure that this is the right behaviour? if we don't have data, why we want to filter for a region with no data?
What if we avoid the click and mouseover on regions with no data at all?

Reply:
this is the behaviour we had before ... and i think it makes sense ...

using makelogs data:
- create region map, select term as geo.src, leave size at 5 ....
- you see the top 5 countries ... however you can now click on something like germany for example (which does not have color applied at the moment as its not top 5) and a filter will apply and you will see the color on germany.